calico_kubernetes/calico_kubernetes.py

```diff
@@ -359,16 +410,72 @@ def _apply_tags(self, profile_name, pod):
             print('Error: Could not apply tags. Profile %s could not be found. Exiting' % profile_name)
             sys.exit(1)
 
-        for k, v in labels.iteritems():
-            tag = '%s_%s' % (k, v)
-            tag = tag.replace('/', '_')
-            print('Adding tag ' + tag)
-            profile.tags.add(tag)
-
+        # Grab namespace and create a tag if it exists.
+        namespace, ns_tag = self._get_namespace_and_tag(pod)
+
+        if ns_tag:
```

Comment:
kubernetes will fill in namespace=default if no namespace is specified. 
Also if namespace is blank for some reason, _get_metadata will issue an warning and return None, which is handled by _get_ns_and_tag